### PR TITLE
implement naming change in sonata integration bundle

### PR DESCRIPTION
**Pull request by ElectricMaxxx**

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
arg StyleCI lives in the wrong year here.

File: app/config/config.yml [hunk 1]
```diff
@@ -122,13 +122,15 @@ cmf_resource_rest:
...
+cmf_sonata_phpcr_admin_integration:
     bundles:
         seo: ~
         menu: ~
         core: ~
         content: ~
-        block: ~
+        block:
+            basepath: /cms/content
+            basepath: /cms/content
```
Comment (wouterj) [inline app/config/config.yml]: duplicate key

File: composer.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
         "symfony-cmf/symfony-cmf": "^2.0",
         "symfony-cmf/seo-bundle": "2.0.*",
         "symfony-cmf/routing-auto-bundle": "2.0.*",
+        "symfony-cmf/sonata-phpcr-admin-integration-bundle": "^1.0",
```
Comment (wouterj) [inline composer.json]: the `sonata-project/doctrine-phpcr-admin-bundle` dependency can now be removed (alongside with the `sonata-project/admin-bundle` dep)

File: app/config/config.yml (hunk 1 above)
Comment (ElectricMaxxx) [inline app/config/config.yml]: yea i meant the `menu_path` for some reasons both are null now.

File: composer.json (hunk 2 above)
Comment (ElectricMaxxx) [inline composer.json]: your comment is on the wrong line, i did the change you requested.

**Comment by ElectricMaxxx** [comment]
@wouterj you have got an idea abaut the one failing build?

**Comment by ElectricMaxxx** [comment]
With the current code i got:
auswahl_079
Seems the generation of the api call isn't right.

**Comment by ElectricMaxxx** [comment]
@wouterj just my current State on that issue (which doesn't belong to that PR, but is important):

When doing the first request `/admin/api/default/?root_node=/cms&amp;_locale=en` i got:

```json
{
  "repository_alias": "default",
  "repository_type": "doctrine_phpcr_odm",
  "payload_alias": null,
  "payload_type": "Doctrine\\ODM\\PHPCR\\Document\\Generic",
  "path": "\/",
  "node_name": false,
  "label": false,
  "repository_path": "\/",
  "children": {
    "cms": {
      "repository_alias": "default",
      "repository_type": "doctrine_phpcr_odm",
      "payload_alias": null,
      "payload_type": "Doctrine\\ODM\\PHPCR\\Document\\Generic",
      "path": "\/cms",
      "node_name": "cms",
      "label": "cms",
      "repository_path": "\/cms",
      "children": {
        "menu": {
          "repository_alias": "default",
          "repository_type": "doctrine_phpcr_odm",
          "payload_alias": null,
          "payload_type": "Doctrine\\ODM\\PHPCR\\Document\\Generic",
          "path": "\/cms\/menu",
          "node_name": "menu",
          "label": "menu",
          "repository_path": "\/cms\/menu",
          "children": {
            "main": []
          }
        },
        "content": {
          "repository_alias": "default",
          "repository_type": "doctrine_phpcr_odm",
          "payload_alias": null,
          "payload_type": "Doctrine\\ODM\\PHPCR\\Document\\Generic",
          "path": "\/cms\/content",
          "node_name": "content",
          "label": "content",
          "repository_path": "\/cms\/content",
          "children": {
            "home": [],
            "company": [],
            "team": [],
            "about": [],
            "contact": [],
            "more": [],
            "projects": [],
            "cmf": [],
            "demo": [],
            "demo_controller": [],
            "demo_template": [],
            "demo_type": [],
            "demo_class": [],
            "publication_demo": [],
            "not_published": [],
            "published_tomorrow": [],
            "singlelocale": [],
            "demo-seo-extractor": [],
            "simple-seo-example": [],
            "simple-seo-property": [],
            "news": []
          }
        },
        "routes": {
          "repository_alias": "default",
          "repository_type": "doctrine_phpcr_odm",
          "payload_alias": null,
          "payload_type": "Doctrine\\ODM\\PHPCR\\Document\\Generic",
          "path": "\/cms\/routes",
          "node_name": "routes",
          "label": "routes",
          "repository_path": "\/cms\/routes",
          "children": {
            "en": [],
            "fr": [],
            "de": [],
            "demo": [],
            "external": [],
            "short": [],
            "short1": [],
            "publicationdemo": [],
            "news": []
          }
        }
      }
    }
  }
}

```

The labe/node_name is `false` and the tree got that node as first node. Didn't find the place where those data goes into the tree yet, but at the end the key of the node is (string) false and that is the part which goes into the routing when fetching the deeper levels. And that node is displayed.

**Comment by ElectricMaxxx** [comment]
Interesting here i got for `/admin/api/default/cms/content/?root_node=/cms&amp;_locale=en` (would be the correct request):
```json
{
  "repository_alias": "default",
  "repository_type": "doctrine_phpcr_odm",
  "payload_alias": null,
  "payload_type": "Doctrine\\ODM\\PHPCR\\Document\\Generic",
  "path": "\/cms\/content\/",
  "node_name": false,
  "label": false,
  "repository_path": "\/cms\/content\/",
  "children": {
    "home": {
      "repository_alias": "default",
      "repository_type": "doctrine_phpcr_odm",
      "payload_alias": null,
      "payload_type": "AppBundle\\Document\\DemoSeoContent",
      "path": "\/cms\/content\/home",
      "node_name": "home",
      "label": "Homepage",
      "repository_path": "\/cms\/content\/home",
      "children": {
        "phpcr_locale:en": {
          "repository_alias": "default",
          "repository_type": "doctrine_phpcr_odm",
          "payload_alias": null,
          "payload_type": "Doctrine\\ODM\\PHPCR\\Document\\Generic",
          "path": "\/cms\/content\/home\/phpcr_locale:en",
          "node_name": "phpcr_locale:en",
          "label": "phpcr_locale:en",
          "repository_path": "\/cms\/content\/home\/phpcr_locale:en",
          "children": []
        },
        "phpcr_locale:fr": {
          "repository_alias": "default",
          "repository_type": "doctrine_phpcr_odm",
          "payload_alias": null,
          "payload_type": "Doctrine\\ODM\\PHPCR\\Document\\Generic",
          "path": "\/cms\/content\/home\/phpcr_locale:fr",
          "node_name": "phpcr_locale:fr",
          "label": "phpcr_locale:fr",
          "repository_path": "\/cms\/content\/home\/phpcr_locale:fr",
          "children": []
        },
        "phpcr_locale:de": {
          "repository_alias": "default",
          "repository_type": "doctrine_phpcr_odm",
          "payload_alias": null,
          "payload_type": "Doctrine\\ODM\\PHPCR\\Document\\Generic",
          "path": "\/cms\/content\/home\/phpcr_locale:de",
          "node_name": "phpcr_locale:de",
          "label": "phpcr_locale:de",
          "repository_path": "\/cms\/content\/home\/phpcr_locale:de",
          "children": []
        },
        "seoMetadata": {
          "repository_alias": "default",
          "repository_type": "doctrine_phpcr_odm",
          "payload_alias": null,
          "payload_type": "Symfony\\Cmf\\Bundle\\SeoBundle\\Doctrine\\Phpcr\\SeoMetadata",
          "path": "\/cms\/content\/home\/seoMetadata",
          "node_name": "seoMetadata",
          "label": "seoMetadata",
          "repository_path": "\/cms\/content\/home\/seoMetadata",
          "children": []
        },
        "additionalInfoBlock": {
          "repository_alias": "default",
          "repository_type": "doctrine_phpcr_odm",
          "payload_alias": null,
          "payload_type": "Symfony\\Cmf\\Bundle\\BlockBundle\\Doctrine\\Phpcr\\ContainerBlock",
          "path": "\/cms\/content\/home\/additionalInfoBlock",
          "node_name": "additionalInfoBlock",
          "label": "additionalInfoBlock",
          "repository_path": "\/cms\/content\/home\/additionalInfoBlock",
          "children": {
            "child1": [],
            "child2": [],
            "child3": []
          },
          "sonata_label": "dashboard.label_container_block",
          "sonata_links": {
            "list": "\/en\/admin\/cmf\/block\/containerblock\/list?id=cms\/content\/home\/additionalInfoBlock",
            "create": "\/en\/admin\/cmf\/block\/containerblock\/create?id=cms\/content\/home\/additionalInfoBlock",
            "batch": "\/en\/admin\/cmf\/block\/containerblock\/batch?id=cms\/content\/home\/additionalInfoBlock",
            "edit": "\/en\/admin\/cmf\/block\/containerblock\/cms\/content\/home\/additionalInfoBlock\/edit",
            "delete": "\/en\/admin\/cmf\/block\/containerblock\/cms\/content\/home\/additionalInfoBlock\/delete",
            "export": "\/en\/admin\/cmf\/block\/containerblock\/export?id=cms\/content\/home\/additionalInfoBlock",
            "show": "\/en\/admin\/cmf\/block\/containerblock\/cms\/content\/home\/additionalInfoBlock\/show"
          }
        },
        "rssBlock": {
          "repository_alias": "default",
          "repository_type": "doctrine_phpcr_odm",
          "payload_alias": null,
          "payload_type": "Symfony\\Cmf\\Bundle\\BlockBundle\\Doctrine\\Phpcr\\RssBlock",
          "path": "\/cms\/content\/home\/rssBlock",
          "node_name": "rssBlock",
          "label": "rssBlock",
          "repository_path": "\/cms\/content\/home\/rssBlock",
          "children": []
        }
      },
      "sonata_label": "dashboard.label_static_content",
      "sonata_links": {
        "list": "\/en\/admin\/app\/demoseocontent\/list?id=cms\/content\/home",
        "create": "\/en\/admin\/app\/demoseocontent\/create?id=cms\/content\/home",
        "batch": "\/en\/admin\/app\/demoseocontent\/batch?id=cms\/content\/home",
        "edit": "\/en\/admin\/app\/demoseocontent\/cms\/content\/home\/edit",
        "delete": "\/en\/admin\/app\/demoseocontent\/cms\/content\/home\/delete",
        "export": "\/en\/admin\/app\/demoseocontent\/export?id=cms\/content\/home",
        "show": "\/en\/admin\/app\/demoseocontent\/cms\/content\/home\/show"
      }
    }
    ...
  }
}
```
(with the changes i made in that PR)

On an other application (without this changes) the enclosing node has the node_name/label `content` but then got a duplication in the tree, which leed to the next falsy api request:
auswahl_080

=> `/admin/api/default/cms/content/content/articles/bla-blubb/?root_node=/cms&amp;_locale=en`

## So ... 

i would say we have two different problems:

- with this PR node_name/label become false for the enclosing node on the api response. some path parameter `null`?
- TreeBrowser handles the enclosing node as a new node, and adds it to the tree

**Comment by ElectricMaxxx** [comment]
Doing:

```
	                    title: false == requestNode.label ? '' : requestNode.label,
	                    key: false == requestNode.node_name ? '' : requestNode.node_name,
``
in web/bundles/cmftreebrowser/js/cmf_tree_browser.fancytree.js:256 solves the `false` problem, but then i also run here into the double node (cms/content/content/..) problem as in my other application.

**Comment by ElectricMaxxx** [comment]
Got a solution, but do not know the way back to the compressed version.

**Comment by wouterj** [comment]
Updated the pull requests. With some more changes in other packages, this should be ready to merge.

This depends on:

 * [ ] https://github.com/sonata-project/SonataDoctrinePhpcrAdminBundle/pull/414
 * [ ] https://github.com/symfony-cmf/tree-browser-bundle/pull/131
 * [ ] https://github.com/symfony-cmf/sonata-phpcr-admin-integration-bundle/pull/77

**Pull request merged**